### DISCUSSION TITLE
How to index files larger than RAM?

### DISCUSSION THREAD

**Discussion by marcin-github** [zeux/qgrep#42] [Q&A] [answered]
Hi,
I play with qgrep to make searching on huge files faster. I just hit problem with memory usage while updating project. I tested on file about 78GB and qgrep didn't fit in RAM (16GB), OOM killed it. Is it possible to index such file? 
Second question, is it possible to deal with compressed text files? Or this is strong no go option?
Thanks

**Comment by zeux** [comment, marked as answer]
Do you know how many lines the file has? I looked into large files before; it’s possible to support but the line counters are 32-bit and on a use case I was aware of before the file had something like 7B lines; handling that required changing file formats and more code than I was comfortable with.

My general advice is to split files like this into more manageable chunks.

Using compressed input files isn’t planned; it would require integrating more external libraries and for anything but single file .gz compression it’s structurally cumbersome to implement, as a single file becomes a folder hierarchy.

**Comment by marcin-github** [comment]
1293634516 lines of json (one json per line)
